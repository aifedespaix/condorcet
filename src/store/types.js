// Misc
export const RESET = "RESET";

// App
export const INITIALIZE = "INITIALIZE";
export const ADD_NOTIFCATION = "ADD_NOTIFCATION";
export const REMOVE_NOTIFCATION = "REMOVE_NOTIFCATION";

// User
export const SET_USER = "SET_USER";
export const CLEAR_USER = "CLEAR_USER";

// Poll
export const SET_POLLS = "SET_POLLS";
export const SET_POLL = "SET_POLL";
export const PUSH_POLL = "PUSH_POLL";
