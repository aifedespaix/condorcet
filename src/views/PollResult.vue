<template>
  <div>
    <h3>Condorcet</h3>

    <Graph v-if="featureFlipping.graphOfTheDuels"></Graph>

    <div v-if="condorcet && featureFlipping.matrixOfTheDuels && answers">
      <table>
        <thead>
          <tr>
            <th>RÃ©ponse</th>
            <th v-for="answer in answers" :key="answer">{{ answer }}</th>
          </tr>
        </thead>
        <tbody>
          <tr v-for="(item, key) in condorcet" :key="key">
            <td>{{ key }}</td>
            <td v-for="item2 in condorcet[key]" :key="`duel-${item2}-${key}`">
              {{ item2 }}
            </td>
          </tr>
        </tbody>
      </table>
      <!-- <div v-for="item in condorcet"></div> -->
    </div>

    <v-list v-if="condorcet && featureFlipping.condorcetRanking">
      <v-list-tile class="panel-block" v-for="item in condorcet" :key="item">
        <p></p>
        {{ item.position }} {{ item.value }}
      </v-list-tile>
    </v-list>

    <div v-if="uninominal && featureFlipping.uninominalRanking">
      <h3>Uninominal</h3>
      <v-list>
        <v-list-tile
          class="list-item"
          v-for="item in uninominal"
          :key="item.position"
        >
          <span class="panel-icon"> {{ item.position }} </span>
          {{ item.value }} - {{ item.numberOfVotes }} voix
        </v-list-tile>
      </v-list>
    </div>
  </div>
</template>

<script>
import { mapState } from "vuex";
import { db } from "@/firebase";
import Graph from "../components/Graph.vue";

export default {
  name: "PollResult",
  components: {
    Graph
  },
  data() {
    return {
      uninominal: null,
      condorcet: null
    };
  },
  computed: {
    ...mapState("app", ["featureFlipping"]),
    answers() {
      return this.condorcet ? Object.keys(this.condorcet) : null;
    }
  },
  async created() {
    const result = await db
      .collection("results")
      .doc(this.$route.params.id)
      .get();

    const { condorcet, uninominal } = result.data();

    this.condorcet = Object.keys(condorcet).reduce((acc, item) => {
      const temp = { ...condorcet[item], [item]: "/" };

      const sortedTemp = Object.keys(temp)
        .sort()
        .reduce((acc, item2) => {
          return { ...acc, [item2]: temp[item2] };
        }, {});

      return { ...acc, [item]: sortedTemp };
    }, {});
    this.uninominal = uninominal;
  }
};
</script>

<style></style>
