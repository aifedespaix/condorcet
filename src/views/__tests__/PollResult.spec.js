import { shallowMount } from "@vue/test-utils";
import PollResult from "../PollResult.vue";
import store from "../../store/index";

jest.mock("../../firebase.js", () => ({
  db: {
    collection: () => ({
      doc: () => ({
        get: () => ({
          data: () => ({
            condorcet: {
              luffy: {
                zorro: 0,
                sanji: 1
              },
              zorro: {
                luffy: 1,
                sanji: 1
              },
              sanji: {
                luffy: 0,
                zorro: 0
              }
            },
            uninominal: [{ value: "luffy", rank: 1 }]
          })
        })
      })
    })
  }
}));

const $route = {
  params: "adz93dze823j"
};

describe("PollResult.vue", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallowMount(PollResult, {
      store,
      mocks: {
        $route
      }
    });
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should set computed props", () => {
    expect(wrapper.vm.answers).toEqual(["luffy", "zorro", "sanji"]);
  });

  it("set data", () => {
    expect(wrapper.vm.uninominal).toEqual([
      {
        rank: 1,
        value: "luffy"
      }
    ]);
  });
});
