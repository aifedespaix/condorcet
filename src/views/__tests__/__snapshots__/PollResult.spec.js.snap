// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollResult.vue should render 1`] = `
<div>
  <h3>Condorcet</h3>
  <!---->
  <div>
    <table>
      <thead>
        <tr>
          <th>RÃ©ponse</th>
          <th>luffy</th>
          <th>zorro</th>
          <th>sanji</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>luffy</td>
          <td>
            /
          </td>
          <td>
            1
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>zorro</td>
          <td>
            1
          </td>
          <td>
            1
          </td>
          <td>
            /
          </td>
        </tr>
        <tr>
          <td>sanji</td>
          <td>
            0
          </td>
          <td>
            /
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <!---->
  <div>
    <h3>Uninominal</h3>
    <v-list-stub>
      <v-list-tile-stub activeclass="primary--text" class="list-item"><span class="panel-icon">  </span>
        luffy - voix
      </v-list-tile-stub>
    </v-list-stub>
  </div>
</div>
`;
